code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -217,6 +256,17 @@ namespace <#= qualifiedNamespace #>
 
                         entityManager.RemoveComponent<NotAuthoritative<<#= componentDetails.TypeName #>>>(entity);
                         entityManager.AddComponentData(entity, new Authoritative<<#= componentDetails.TypeName #>>());
+
+                        // Add event senders
+<# foreach (var eventDetail in eventDetailsList) { #>
+                        {
```

Comment:
Are these block here to reuse local variable names?

Reply:
As far as I can tell eventSender is the only name which is reused. That could be named something according to what it does. E.g eventSenderToAdd, eventSenderToRemove.

Reply:
You'd need to generate `eventSender<#= eventName #>` everywhere its used. I don't think scoping things to avoid awkward naming is a problem

Reply:
Not really a big deal as this doesn't affect readability much and we can easily change it later if needs be.